### ISSUE TITLE
Power OpenBLAS 0.2.20 Segmentation fault in  dgemm_beta

### ISSUE THREAD

**Issue by sh1ng** [xianyi/OpenBLAS#2307]
```
#0  0x00002000093f5270 in dgemm_beta ()
   from ....libopenblas.so.0
```

```
   ┌───────────────────────────────────────────────────────────────────────────┐
   │0x2000093f525c <dgemm_beta+156> stfd    f0,88(r7)                          │
   │0x2000093f5260 <dgemm_beta+160> stfd    f0,96(r7)                          │
   │0x2000093f5264 <dgemm_beta+164> stfd    f0,104(r7)                         │
   │0x2000093f5268 <dgemm_beta+168> stfd    f0,112(r7)                         │
   │0x2000093f526c <dgemm_beta+172> stfd    f0,120(r7)                         │
  >│0x2000093f5270 <dgemm_beta+176> dcbst   r6,r7                              │
   │0x2000093f5274 <dgemm_beta+180> addi    r7,r7,128                          │
   │0x2000093f5278 <dgemm_beta+184> bdnz    0x2000093f5230 <dgemm_beta+112>    │
   │0x2000093f527c <dgemm_beta+188> ori     r2,r2,0                            │
   │0x2000093f5280 <dgemm_beta+192> andi.   r0,r3,15                           │
   │0x2000093f5284 <dgemm_beta+196> mtctr   r0                                 │
   │0x2000093f5288 <dgemm_beta+200> beq     0x2000093f52a0 <dgemm_beta+224>    │
   │0x2000093f528c <dgemm_beta+204> ori     r2,r2,0                            │
   └───────────────────────────────────────────────────────────────────────────┘
multi-thre Thread 0x20000 In: dgemm_beta          Line: ??   PC: 0x2000093f5270 
```

OpenBLAS is built from sources with `make CBLAS_ONLY=1 && make PREFIX=/usr/local/openblas install`

Probably the issue occurs on Power9, but I'm waiting confirmation from our support team.

**Comment by martin-frbg** [comment]
Any particular reason why you would want to use 0.2.20 from two years ago, rather than the current 0.3.7 ?

**Comment by sh1ng** [comment]
We just got it from our customer, I'm going to upgrade to 0.3.7 but it'd be nice to get a confirmation that the issue is fixed or something. 

Is it related to https://github.com/xianyi/OpenBLAS/pull/2013 ?

**Comment by martin-frbg** [comment]
Very probably yes. At least that it fails at what looks iike the very same dcbst instruction is very suggestive.

**Comment by sh1ng** [comment]
Taking into account https://github.com/xianyi/OpenBLAS/issues/2254 have you thought to back-port the fix and bump v0.2.21?

**Comment by martin-frbg** [comment]
v0.2.x is pretty much dead - all development during the past two years has taken place on the 0.3 branch, a side effect of the effective change of (active) maintainership as xianyi is pursuing other projects. (There was no significant "technical" change between 0.2.20 and 0.3.0 that would have required a jump in version number, more like a safeguard against accidentally trashing the repository)

**Comment by martin-frbg** [comment]
Are you also limited to gcc 4.x ? (regarding #2254 and its various workarounds for old gcc versions)

**Comment by sh1ng** [comment]
Using not a default compiler requires additional testing and complicates building. As result it will get into production much latter.

**Comment by martin-frbg** [comment]
There is two years worth of bug fixes in 0.3.7, including things that specifically affected power8 (not least several wrong or missing declarations in the assembly kernels), and #2254 is originally/mostly about old bugs in that default compiler. That said there should be a 0.3.8 release with the accumulated fixes and workarounds from #2254 in the near future.
